Framework/DataHandling/src/LoadILLPolarizedDiffraction.cpp

```diff
@@ -0,0 +1,411 @@
+    const API::MatrixWorkspace_sptr &workspace, const NXEntry &entry,
+    int bankId) {
+
+  std::vector<double> twoTheta(static_cast<int>(D7_NUMBER_PIXELS_BANK));
+
+  if (getPropertyValue("PositionCalibration").compare("Nexus") == 0) {
+    NXFloat twoThetaPixels = entry.openNXFloat(
+        "D7/Detector/bank" + std::to_string(bankId) + "_offset");
+    twoThetaPixels.load();
+    for (auto pixelId = 0; pixelId < twoThetaPixels.size(); pixelId++) {
+      twoTheta[pixelId] = twoThetaPixels[pixelId];
+    }
+  } else {
+    IAlgorithm_sptr loadInst = createChildAlgorithm("LoadParameterFile");
+    loadInst->setPropertyValue("Filename", getPropertyValue("YIGFilename"));
+    loadInst->setProperty("Workspace", workspace);
+    loadInst->execute();
+
+    auto instrumentMap = workspace->instrumentParameters();
+    Instrument_const_sptr instrument = workspace->getInstrument();
+
+    for (auto ii = 0; ii < static_cast<int>(D7_NUMBER_PIXELS_BANK); ii++) {
+      auto pixel = instrument->getDetector(ii + 1);
+      std::vector<double> twoThetaRead =
+          (pixel->parameterMap())
+              .getDouble(pixel->getName(),
+                         std::string("twoTheta_bank" + std::to_string(bankId)));
+      twoTheta[ii] = twoThetaRead[0];
+    }
+  }
+  return twoTheta;
+}
+
+/**
+ * Rotates each pixel to its corresponding 2theta read from the file
+ * @param entry : entry from which the 2theta positions will be read
+ * @param workspace : workspace containing the instrument being moved
+ */
+void LoadILLPolarizedDiffraction::moveTwoTheta(
+    const NXEntry &entry, API::MatrixWorkspace_sptr &workspace) {
+
+  Instrument_const_sptr instrument = workspace->getInstrument();
+
+  auto &componentInfo = workspace->mutableComponentInfo();
+  for (auto ii = 0; ii < static_cast<int>(D7_NUMBER_BANKS); ii++) {
+    NXFloat twoThetaBank = entry.openNXFloat(
+        "D7/2theta/actual_bank" +
+        std::to_string(ii + 2)); // detector bank IDs start at 2
+    twoThetaBank.load();
+    if (getPropertyValue("PositionCalibration").compare("None") == 0) {
+      Quat rotation(-twoThetaBank[0], V3D(0, 1, 0));
+      IComponent_const_sptr currentBank = instrument->getComponentByName(
+          std::string("bank" + std::to_string(ii + 2)));
+      const auto componentIndex =
+          componentInfo.indexOf(currentBank->getComponentID());
+      componentInfo.setRotation(componentIndex, rotation);
+    } else {
+      std::vector<double> twoThetaPixels =
+          loadTwoThetaDetectors(workspace, entry, ii + 2);
+      for (auto jj = 0; jj < static_cast<int>(D7_NUMBER_PIXELS_BANK); jj++) {
+        IComponent_const_sptr pixel = instrument->getDetector(
+            ii * static_cast<int>(D7_NUMBER_PIXELS_BANK) + jj + 1);
+        const auto pixelIndex = componentInfo.indexOf(pixel->getComponentID());
+        V3D position = pixel->getPos();
+        double const radius = sqrt(pow(position[0], 2) + pow(position[2], 2));
+        position = V3D(
```

Comment:
Since you are just rotating, this would look nicer in polar coordinates.

Reply:
I completely agree, originally I didn't find the right functions but it is going to be corrected.